llvm/lib/Analysis/TargetTransformInfo.cpp

```diff
@@ -1419,6 +1419,10 @@ unsigned TargetTransformInfo::getMaxNumArgs() const {
   return TTIImpl->getMaxNumArgs();
 }
 
+bool TargetTransformInfo::canVectorizei8s() const {
```

Comment:
Not sure about this TTI hook. In general, there is no special about i8s. They just happen to require certain special handling on AMDGPUs. However, for other targets, i8s may be first class citizens / legal, so they are already legalized by default -- x86 for example I believe has legal i8s

Reply:
Right, I think the name of that hook can be improved. The word vectorize in there is not quite correct.